pkg/reconciler/trigger/trigger.go

```diff
@@ -207,6 +240,37 @@ func (r *Reconciler) reconcile(ctx context.Context, t *v1alpha1.Trigger) error {
 	return nil
 }
 
+func (r *Reconciler) triggerMarkedReadyByDependency(dependencyObjRef corev1.ObjectReference, ctx context.Context, t *v1alpha1.Trigger) error {
+	gvk := schema.GroupVersionKind{
+		Group:   dependencyObjRef.GroupVersionKind().Group,
+		Version: dependencyObjRef.GroupVersionKind().Version,
+		Kind:    dependencyObjRef.Kind,
+	}
+	gvr, _ := meta.UnsafeGuessKindToResource(gvk)
+	_, lister, err := r.buildInformerFactory.Get(gvr)
+	if err != nil {
+		logging.FromContext(ctx).Error(fmt.Sprintf("Error getting a lister for a resource with gvk '%+v', gvr '%+v'", gvk, gvr), zap.Error(err))
+		t.Status.MarkDependencyReadyUnknown("FailedToListResource", "Failed to list resource")
```

Comment:
The message should probably include the error string:

```go
t.Status.MarkDependencyReadyUnknown("FailedToListResource", "Failed to list resource: %v", err)

Reply:
I added the detailed error msg into the logging context. When I describe the trigger, it will list the detailed the error msg. I followed the code style when trigger tries to reconcile broker and subscription. It seems like for the message when marking status, it's pretty short. We added the detailed error msg in logging. Should we change it?

Reply:
FYI: This is the example of using reason and msg when marking status https://github.com/knative/serving/blob/master/docs/spec/errors.md#condition-reason-and-message

Reply:
I think we should add the error to the status message. Most users will not be looking in our reconciler logs. And I think no users _want_ to look through our reconciler logs. Therefore, as I user, I want the error message in the object's status.